### PR TITLE
Finished single class booking-payment integration

### PR DESCRIPTION
**Pull request by chenjoe569**
Got rid of the old credit booking system. Integrated stripe payment with the single class booking workflow. Basically, you confirm to pay after booking for one class (one hour). Fixed some bugs and edited tests.

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def index
             @calendars = Calendar.all.where("start_time > ?", Time.now.beginning_of_day).order(:start_time)
         #add admin
         end
-         @calendarsShow = @calendars.limit(5)
+        @calendarsShow = @calendars.limit(5)
```
Comment (zdehkordi) [inline app/controllers/home_controller.rb]: thank you for making code look nice for golden repo

File: app/views/charges/checkout.html.erb [hunk 2]
```diff
@@ -6,20 +6,25 @@
...
-  <br> <%= link_to "Back to Cart", new_charge_path, :class => 'navbar-link'  %>
+  <br> <%= link_to "Back to Booking", booking_path, :class => 'navbar-link'  %>
   </p>
   </article>
   <article>
   <%= hidden_field_tag(:amount, @amount) %>
-  <%= hidden_field_tag(:group_num_credit, @group_num_credit) %>
-  <%= hidden_field_tag(:assigned_num_credit, @assigned_num_credit) %>
-  <%= hidden_field_tag(:custom_num_credit, @custom_num_credit) %>
+  <%= hidden_field_tag(:event_start, @event_start) %>
```
Comment (zdehkordi) [inline app/views/charges/checkout.html.erb]: More transitioning from credits to direct payments

File: app/views/charges/create.html.erb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 <h2>Thanks, you paid <strong> <%= number_to_currency(@amount_in_create) %></strong>!</h2>
-<br> <%= link_to "Back to My Profile Page", member_profile_path %>
+<p>Thank you for booking a lesson!<br /></p>
+<%= link_to 'Book another lesson', booking_path %>
+<br> <%= link_to "Back to Home Page", root_path %>
```
Comment (zdehkordi) [inline app/views/charges/create.html.erb]: We can make these buttons in the CSS branch, I'll work on making it look pretty

File: app/controllers/charges_controller.rb [hunk 4]
```diff
@@ -64,35 +61,55 @@ def create
...
+      if (params[:user].nil? || params[:user][:temp_availability].nil?)
+        flash[:alert] = "Please choose a time slot."
+        redirect_to :controller => "user", :action => "booking" 
+        return
+      end
+
+      # Parsing time
+      start_time = DateTime.parse(params[:user][:temp_availability].split(',')[0])
+      end_time = DateTime.parse(params[:user][:temp_availability].split(',')[1])
+      @event_start = DateTime.new(DateTime.now.year.to_i, params[:month].to_i, params[:day].to_i, start_time.hour, start_time.minute, 0, "-07:00")
```
[the remark this answers is not in the document]
Comment (chenjoe569) [reply app/controllers/charges_controller.rb]: The one he had earlier didn't work, but this does. I can look more into it, but for now I think this is ok. There are more important things we should get to. I'll look more into it later

File: app/views/charges/create.html.erb (hunk 3 above)
Comment (chenjoe569) [reply app/views/charges/create.html.erb]: Thanks!

File: app/controllers/charges_controller.rb [hunk 5]
```diff
@@ -64,35 +61,55 @@ def create
...
-      @custom_num_credit = params[:custom_num_credit].to_i + current_user.custom_num_credit.to_i
-      @group_num_credit = params[:group_num_credit].to_i + current_user.group_num_credit.to_i
-      @assigned_num_credit = params[:assigned_num_credit].to_i + current_user.assigned_num_credit.to_i
-      current_user.custom_num_credit = @custom_num_credit.to_s
-      current_user.group_num_credit = @group_num_credit.to_s
-      current_user.assigned_num_credit = @assigned_num_credit.to_s
-      current_user.save!
-     
+      # Storing booked lesson in the database
+      event_start_time = params[:event_start_time]
```
[the remark this answers is not in the document]
Comment (chenjoe569) [reply app/controllers/charges_controller.rb]: What do you mean two events?

[the remark this answers is not in the document]
Comment (chenjoe569) [reply app/controllers/charges_controller.rb]: There is only one event, but one records start time and end time. So yes we need both

**Pull request merged**